webapp/sources/rudder/rudder-core/src/test/resources/configuration-repository/techniques/ncf_techniques/technique_by_Rudder/1.0/technique.ps1

```diff
@@ -7,50 +7,98 @@
+    TechniqueName = $techniqueName
+  }
+  $class = [Rudder.Condition]::canonify("(debian)")
+  if ($localContext.Evaluate($class)) {
+    $call = Package-Install-Version -PackageName "$($node.properties[apache_package_name])" -PackageVersion "2.2.11" -PolicyMode: $policyMode
+    $methodContext = Compute-Method-Call @reportParams -MethodCall $call
+    $localContext.merge($methodContext)
   } else {
-    _rudder_common_report_na -componentName "Customized component" -componentKey "$($node.properties[apache_package_name])" -message "Not applicable" -Report:$true -reportId $reportId -techniqueName $techniqueName -auditOnly:$auditOnly
+    Rudder-Report-NA @reportParams
   }
 
   $reportId=$reportIdBase+"id2"
-
-  $class = "(debian.windows)"
-  if (Evaluate-Class $class $local_classes $system_classes) {
-    $local_classes = Merge-ClassContext $local_classes $(Command-Execution -Command "Write-Host `"testing special characters ` è &é 'à é `"" -componentName "Command execution" -Report:$false -reportId $reportId -techniqueName $techniqueName -auditOnly:$auditOnly).get_item("classes")
+  $reportParams = @{
+    ClassPrefix = "/!\ a definir /!\"
+    ComponentKey = "Write-Host `"testing special characters ` è &é 'à é `""
+    ComponentName = "Command execution"
+    PolicyMode = $policyMode
+    ReportId = $reportId
+    DisableReporting = $true
+    TechniqueName = $techniqueName
+  }
+  $class = [Rudder.Condition]::canonify("(debian.windows)")
+  if ($localContext.Evaluate($class)) {
+    $call = Command-Execution -Command "Write-Host `"testing special characters ` è &é 'à é `"" -PolicyMode: $policyMode
+    $methodContext = Compute-Method-Call @reportParams -MethodCall $call
+    $localContext.merge($methodContext)
   } else {
-    _rudder_common_report_na -componentName "Command execution" -componentKey "Write-Host `"testing special characters ` è &é 'à é `"" -message "Not applicable" -Report:$false -reportId $reportId -techniqueName $techniqueName -auditOnly:$auditOnly
+    Rudder-Report-NA @reportParams
   }
 
   $reportId=$reportIdBase+"id3"
-
-  $class = "package_install_version_" + $(Canonify-Class $($node.properties[apache_package_name])) + "_repaired"
-  if (Evaluate-Class $class $local_classes $system_classes) {
-    $local_classes = Merge-ClassContext $local_classes $(Service-Start -ServiceName "$($node.properties[apache_package_name])" -componentName "Customized component" -Report:$true -reportId $reportId -techniqueName $techniqueName -auditOnly:$auditOnly).get_item("classes")
+  $reportParams = @{
+    ClassPrefix = "/!\ a definir /!\"
+    ComponentKey = "$($node.properties[apache_package_name])"
+    ComponentName = "Customized component"
+    PolicyMode = $policyMode
+    ReportId = $reportId
+    DisableReporting = $false
+    TechniqueName = $techniqueName
+  }
+  $class = [Rudder.Condition]::canonify("package_install_version_" + $(Canonify-Class $($node.properties[apache_package_name])) + "_repaired")
```

Comment:
Use `[Rudder.Condition]::Canonify()` when you want to canonify something starting 7.2